google/auth/_default.py

```diff
@@ -234,13 +243,22 @@ def default():
     order:
 
     1. If the environment variable ``GOOGLE_APPLICATION_CREDENTIALS`` is set
-       to the path of a valid service account private key file, then it is
+       to the path of a valid service account JSON private key file, then it is
        loaded and returned. The project ID returned is the project ID defined
-       in the service account file.
-    2. If the `Google Cloud SDK`_ is installed and has credentials set via
-       ``gcloud auth application-default login`` they are loaded and returned.
-       The Project ID is the project ID configured with ``gcloud init`` or
-       ``gcloud config set project``.
+       in the service account file if available (some older files do not
+       contain project ID information).
+    2. If the `Google Cloud SDK`_ is installed and has application default
+       credentials set they are loaded and returned.
+
+       To enable application default credentials with the Cloud SDK run::
+
+            gcloud auth application-default login
+
+       If the Cloud SDK has an active project, the project ID is returned. The
+       active project can be set using::
+
+            gcloud config set project
```

Comment:
Yeah, it's a way more roundabout way of setting the project. Users typically only run it once.